### PR TITLE
Remove workarround restart manager logtest

### LINKED ISSUE #1506: Remove workarround to restart manager in the logtest IT
Hi team!,

The new logtest functionality allows new files to be loaded at runtime. To optimize and purchase this, it is necessary to remove the workaround of invalid decoder and rule tests.

Regards,
Julian

### PR DESCRIPTION
**Pull request by juliancnn**
|Related issue|
|---|
| Closes #1506 |

Hi team!,

This PR implements 3 changes:
- [x] Remove workarround to restart manager in the tests: test_invalid_rule_decoders_syntax
- [x] Remove workarround to restart manager in the tests: test_rules_decoders_load
- [x] Change the debug mode of analysisd before running remove session tests, (no need to set it manually anymore, neither in jenkis).

Regards,
Julian

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [x] `provision_documentation.sh` generate the docs without errors.

### CODE REVIEW

**Comment by JcabreraC** [comment]
## Test

### test_invalid_decoder_syntax.py
```
# python3 -m pytest tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py -vvs
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.9', 'Platform': 'Linux-5.4.0-77-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'pytest': '6.2.3', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '1.9.0'}}
rootdir: /home/Wazuh/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-1.9.0
collected 11 items                                                                                  

tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: garbage file] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: no closing XML tag] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: no existing parent] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: no existing attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: decoder with no name] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: regex attribute without order attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: regex attribute without prematch/program_name/parent attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: order attribute without regex attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: two-level order parenting] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: invalid plugin_decoder] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_decoder_syntax.py::test_invalid_decoder_syntax[Invalid decoder syntax: invalid offset] PASSED

======================================== 11 passed in 0.58s =========================================
```

### test_invalid_rules_syntax.py
```
# python3 -m pytest tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py -vvs
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.9', 'Platform': 'Linux-5.4.0-77-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'pytest': '6.2.3', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '1.9.0'}}
rootdir: /home/Wazuh/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-1.9.0
collected 15 items                                                                                  

tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: garbage file] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: no closing XML tag] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: no group] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: invalid level] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: invalid rule overwrite] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: same_* attribute without frequency and timeframe] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: invalid weekday] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: no existing if_sid rule number] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: no existing if_matched_sid rule number attribute without frequency and timeframe] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: non existing/invalid attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: no invalid rule label] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: non existing group in if_group] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: invalid values on option attribute] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: different_* attribute without frequency and timeframe] PASSED
tests/integration/test_logtest/test_invalid_rule_decoders_syntax/test_invalid_rules_syntax.py::test_invalid_rule_syntax[Invalid rules syntax: non existing decoder] PASSED

======================================== 15 passed in 6.29s =========================================
```

### test_remove_old_sessions.py
```
# python3 -m pytest tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py -vvs
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.9', 'Platform': 'Linux-5.4.0-77-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'pytest': '6.2.3', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '1.9.0'}}
rootdir: /home/Wazuh/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-1.9.0
collected 1 item                                                                                    

tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py::test_remove_old_session[get_configuration0] PASSED2021-06-30T13:11:52.1625051512


======================================== 1 passed in 53.17s =========================================
```


### test_remove_old_session_for_inactivity.py
```
python3 -m pytest tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_session_for_inactivity.py -vvs
======================================== test session starts ========================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.9', 'Platform': 'Linux-5.4.0-77-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'pytest': '6.2.3', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '1.9.0'}}
rootdir: /home/Wazuh/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-1.9.0
collected 1 item                                                                                    

tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_session_for_inactivity.py::test_remove_old_session_for_inactivity[get_configuration0] PASSED2021-06-30T12:28:37.1625048917


======================================== 1 passed in 57.67s =========================================
```

## Documentation

**Comment by juliancnn** [comment]
LGTM!!

**Review by nmkoremblum — DISMISSED** [review]
:+1:

**Comment by JcabreraC** [comment]
LGTM !

**Review by JcabreraC — DISMISSED** [review]

File: tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_session_for_inactivity.py [hunk 1]
```diff
@@ -42,7 +51,8 @@ def get_configuration(request):
 
 
 # Test
-def test_remove_old_session_for_inactivity(get_configuration, configure_environment,
+def test_remove_old_session_for_inactivity(get_local_internal_options, configure_local_internal_options,
+                                           get_configuration, configure_environment,
                                            restart_wazuh, connect_to_sockets_function):
```
Comment (juliamagan) [inline tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_session_for_inactivity.py:56]: After talking with @snaow we have considered that new development or changes into existing tests should follow new repository standards about Wazuh daemons. For now, it is required to use fixtures to start only required daemons, and turn them off when the test is over. That means, that the default status of Wazuh would be off and each test should be responsible for daemons handling.
The reasons for this change are:

- Faster testing
- More encapsulated tests

File: tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py [hunk 2]
```diff
@@ -48,7 +57,8 @@ def get_configuration(request):
 
 
 # Test
-def test_remove_old_session(get_configuration, configure_environment, restart_wazuh):
+def test_remove_old_session(get_local_internal_options, configure_local_internal_options,
+                            get_configuration, configure_environment, restart_wazuh):
```
Comment (juliamagan) [inline tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py:61]: Same as before

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py [hunk 3]
```diff
@@ -22,17 +24,23 @@
...
 
 # Variables
+logtest_startup_timeout = 30
 wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
-logtest_sock = os.path.join(os.path.join(WAZUH_PATH, 'queue', 'sockets', 'logtest'))
-
+local_internal_options = {'analysisd.debug': str(1)}
 msg_create_session = """{"version":1, "command":"log_processing", "parameters":{
 "event": "Oct 15 21:07:56 linux-agent sshd[29205]: Invalid user blimey from 18.18.18.18 port 48928",
 "log_format": "syslog", "location": "master->/var/log/syslog"}}"""
```
Comment (Rebits) [inline tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py:33]: Use `json.dump` instead
```suggestion
create_session_data = {'version': 1, 'command': 'log_processing', 'parameters': {'event': 'Oct 15 21:07:56 linux-agent sshd[29205]: Invalid user blimey from 18.18.18.18 port 48928', 'log_format': 'syslog', 'location': 'master->/var/log/syslog'}}
msg_create_session = json.dumps(create_session_data)


```

File: tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py [hunk 4]
```diff
@@ -22,17 +24,23 @@
 configurations = load_wazuh_configurations(configurations_path, __name__)
 
 # Variables
+logtest_startup_timeout = 30
```
Comment (Rebits) [inline tests/integration/test_logtest/test_remove_old_sessions/test_remove_old_sessions.py:27]: Maybe 30 seconds is too much to wait for logtest. Maybe is better to set it to '3'.

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ, but some changes are required.

**Comment by Rebits** [comment]
Other minor considerations about this PR:

- It should be desirable to replace the not very informative error message ' Event not found' with a more precise one
```
           wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
                                    callback=callback_session_initialized,
                                    error_message='Event not found')
```
- Change test configuration file (`wazuh_conf.yaml`) to a more precise one
- Use `file_monitoring`, master branch fixture. Tets header should be:
```
def test_remove_old_session(configure_local_internal_options_module,
                            get_configuration, configure_environment, file_monitoring,
                            restart_required_logtest_daemons,
                            wait_for_logtest_startup):

```

Also `wazuh_log_monitor` should be renamed to log_monitor in order to make this fixture work.

**Comment by juliamagan** [comment]
### Package
| Version | Revision | Link|
|---|---|---|
|4.3.0|40301|http://packages-dev.wazuh.com/warehouse/pullrequests/4.3/rpm/var/wazuh-manager-4.3.0-0.commit0a7ba8b.x86_64.rpm



## Testing

### Logtest

|  OS  | Local   | Jenkins     | Notes
|---    |---    |---    |---    |
| PS1   | |   | |
| PS2   |   |  | |
| PS3   |    |   | |

* * * 

- :green_circle:: All pass
- :yellow_circle:: Some warnings
- :red_circle:: Some errors/fails
- :large_blue_circle:: In progress

**Pull request merged**